Source/GoBot/FeedStrategy/RecentlyActivePostsAndContactsAlgorithm.swift

```diff
@@ -0,0 +1,409 @@
+//
+//  PostsAndContactsAlgorithm.swift
+//  Planetary
+//
+//  Created by Martin Dutra on 12/4/22.
+//  Copyright © 2022 Verse Communications Inc. All rights reserved.
+//
+
+import Foundation
+import SQLite
+
+/// This algorithm returns a feed with user's and follows' posts, and follows' following other users in the network.
+/// If a root post is replied to it will be boosted back up to the top of the feed.
+///
+/// It doesn't include pubs follows and posts in the feed.
+///
+/// NOTE: This has a lot of code copied from PostsAndContactsAlgorithm. We should factor it out. (#564)
+class RecentlyActivePostsAndContactsAlgorithm: NSObject, FeedStrategy {
+
+    // swiftlint:disable indentation_width
+    private let countNumberOfKeysQuery = """
+        SELECT COUNT(*)
+        FROM messages
+        JOIN authors ON authors.id == messages.author_id
+        LEFT JOIN posts ON messages.msg_id == posts.msg_ref
+        LEFT JOIN contacts ON messages.msg_id == contacts.msg_ref
+        LEFT JOIN abouts AS contact_about ON contact_about.about_id == contacts.contact_id
+        LEFT JOIN authors AS contact_author ON contact_author.id == contacts.contact_id
+        WHERE messages.type IN ('post', 'contact')
+        AND messages.is_decrypted == false
+        AND messages.hidden == false
+        AND (type <> 'post' OR posts.is_root == true)
+        AND (type <> 'contact'  OR (
+                contact_about.about_id IS NOT NULL AND contact_author.author NOT IN (SELECT key FROM pubs)
+            ))
+        AND (authors.author IN (
+                SELECT authors.author FROM contacts
+                JOIN authors ON contacts.contact_id == authors.id
+                WHERE contacts.author_id = ? AND contacts.state == 1
+            ) OR authors.id == ?)
+        AND authors.author NOT IN (SELECT key FROM pubs)
+        AND claimed_at < ?;
+    """
+    // swiftlint:enable indentation_width
+
+    // swiftlint:disable indentation_width
+    private let fetchKeysQuery = """
+        SELECT messagekeys.key,
+               (SELECT COALESCE(tangled_message.claimed_at, messages.claimed_at)
+                FROM tangles
+                JOIN messages AS tangled_message ON tangles.msg_ref == tangled_message.msg_id
+                WHERE tangles.root == messages.msg_id
+                /*WHERE tangled_message.claimed_at < ?*/
```

Comment:
Remove this comment

Reply:
Oops, actually I wanted to uncomment this to filter out replies from the future. It's done now.